### DISCUSSION TITLE
Non-nullable field with default value

### DISCUSSION THREAD

**Discussion by thaindq** [mikro-orm/mikro-orm#2255] [Q&A] [answered]
I have an entity with `Timestamp` field like this
```
    @Property({
        fieldName: 'Timestamp',
        defaultRaw: 'current_timestamp',
    })
    timestamp!: Date;
```
But when I insert a new record into the table, since that field is left `undefined` so it is translated as `NULL` in the query resulting this error
```
"insert into \"Customer\" (\"Email\", \"Name\", \"Password\", \"Timestamp\") values ('test@gmail.com', 'test', '', NULL) returning \"Id\", \"Timestamp\" - null value in column \"Timestamp\" of relation \"Customer\" violates not-null constraint",
```

Is there a way to tell the orm to skip the `value` for that timestamp column? I'm expecting it to leave that field out of the query and let the database handle the default value since we have `defaultRaw` defined.

**Comment by B4nan** [comment, marked as answer]
> But when I insert a new record into the table

How do you do that? It might be TS misconfiguration, as it should not cause any problems. I know that with `esnext` the class properties are compiled badly (for this usecase) - they produce the value `undefined` and the property is iterable. 

Sounds a bit like #1984 - see [this comment](https://github.com/mikro-orm/mikro-orm/issues/1984#issuecomment-898888483)

**Reply by thaindq** [reply]
Thank you, yeah it's the same issue. Disabling `useDefineForClassFields` in `tsconfig.json` fixes it.
